.github/workflows/Func_Env_Build_Test_CI.yml

```diff
@@ -14,19 +14,85 @@ concurrency:
+    name: unittest
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.6', '3.7', '3.8', '3.9', '3.10' ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest pytest-cov
+          if [[ -f requirements.txt ]]; then pip install -r requirements.txt; fi
+          if [[ -f tests_requirements.txt ]]; then pip install -r tests_requirements.txt; fi
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+      - name: ⚙️ Set Kubeconfig and hosts
+        env:
+          KUBECONFIG: ${{ secrets.FUNC_KUBECONFIG }}
+          RUNNER_PATH: ${{ secrets.RUNNER_PATH }}
+          OCP_HOSTS: ${{ secrets.FUNC_OCP_HOSTS }}
+        run: |
+          mkdir -p "$RUNNER_PATH/.kube/"
+          echo "$KUBECONFIG" > "$RUNNER_PATH/.kube/config"
+          echo "KUBECONFIG_PATH=$RUNNER_PATH/.kube/config" >> "$GITHUB_ENV"
+          sudo tee -a /etc/hosts <<< "$OCP_HOSTS" > /dev/null
+      - name: 📃 Unittest with pytest
+        env:
+          KUBEADMIN_PASSWORD: ${{ secrets.FUNC_KUBEADMIN_PASSWORD }}
+          PIN_NODE1: ${{ secrets.FUNC_PIN_NODE1 }}
+          PIN_NODE2: ${{ secrets.FUNC_PIN_NODE2 }}
+          RUNNER_PATH: ${{ secrets.RUNNER_PATH }}
+          ELASTICSEARCH: ${{ secrets.FUNC_ELASTICSEARCH }}
+          ELASTICSEARCH_PORT: ${{ secrets.FUNC_ELASTICSEARCH_PORT }}
+          ELASTICSEARCH_USER: ${{ secrets.FUNC_ELASTICSEARCH_USER }}
+          ELASTICSEARCH_PASSWORD: ${{ secrets.FUNC_ELASTICSEARCH_PASSWORD }}
+          IBM_REGION_NAME: ${{ secrets.IBM_REGION_NAME }}
+          IBM_ENDPOINT_URL: ${{ secrets.IBM_ENDPOINT_URL }}
+          IBM_ACCESS_KEY_ID: ${{ secrets.IBM_ACCESS_KEY_ID }}
+          IBM_SECRET_ACCESS_KEY: ${{ secrets.IBM_SECRET_ACCESS_KEY }}
+          IBM_BUCKET: ${{ secrets.IBM_BUCKET }}
+          IBM_KEY: ${{ secrets.IBM_KEY }}
+          RUN_ARTIFACTS_URL: ${{ secrets.FUNC_RUN_ARTIFACTS_URL }}
+        run: |
+          # Install Dockerfile content for pytest
+          # install oc/kubectl
+          oc_version=4.11.0-0.okd-2022-10-15-073651
+          curl -L https://github.com/openshift/okd/releases/download/${oc_version}/openshift-client-linux-${oc_version}.tar.gz -o $RUNNER_PATH/openshift-client-linux-${oc_version}.tar.gz
+          tar -xzvf $RUNNER_PATH/openshift-client-linux-${oc_version}.tar.gz -C $RUNNER_PATH/
```

Comment:
Don't need the temporary file:

```curl -L "https://github.com/openshift/okd/releases/download/${oc_version}/openshift-client-linux-${oc_version}.tar.gz" | tar xvf - -C "$RUNNER_PATH/"```

Reply:
its running inside Ubuntu container, working for me, prefer not to change it

Reply:
This isn't that big of a deal, but it is not necessary and if the tar fails the file will be left around (and yes, the container exit will clean it up, but it's best not to leave things like that around).

Reply:
Tried it on my local and it failed:
```
$ curl -L "https://github.com/openshift/okd/releases/download/4.11.0-0.okd-2022-10-15-073651/openshift-client-linux-4.11.0-0.okd-2022-10-15-073651.tar.gz" | tar xvf - -C "/tmp/"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0tar: Archive is compressed. Use -z option
tar: Error is not recoverable: exiting now
  0 50.1M    0 17471    0     0   5823      0  2:30:25  0:00:03  2:30:22 32234
curl: (23) Failed writing body (335 != 1422)
```